xdem/spstats.py

```diff
@@ -9,19 +9,123 @@
 import random
 import warnings
 from functools import partial
-from typing import Callable, Union
+from typing import Callable, Union, Iterable, List, Optional, Sequence
 
+import itertools
 import matplotlib.pyplot as plt
+from numba import njit
 import numpy as np
 import pandas as pd
 from scipy import integrate
 from scipy.optimize import curve_fit
+from scipy.stats import binned_statistic, binned_statistic_2d, binned_statistic_dd
+from xdem.spatial_tools import nmad
 
 with warnings.catch_warnings():
     warnings.filterwarnings("ignore", category=DeprecationWarning)
     import skgstat as skg
     from skgstat import models
 
+def nd_binning_scipy(dh: np.ndarray, list_var: List[np.ndarray], list_var_names=List[str], list_var_bins: Optional[List[Iterable]] = None,
+                     statistics: list[Union[str, Callable, None]] = ['count', np.nanmedian ,nmad], list_ranges : Optional[List[Sequence]] = None) \
+        -> tuple[list[pd.DataFrame],list[pd.DataFrame],pd.DataFrame]:
+    """
+    Quantify non-stationarities in elevation measurement errors with one or several, terrain or instrument-related,
+    explanatory variables. See documentation of scipy.stats.binned_statistic_dd for more details.
+
+    :param dh: elevation differences
+    :param list_var: list of explanatory variables
+    :param list_var_names: list of names of the explanatory variables
+    :param list_var_bins: list of bins for the explanatory variables
```

Comment:
You should state what the default is (10 bins from min to max), as it is not clear from the docstring.

Reply:
Added